### PR TITLE
Add QuickStart Guide and Improve Documentation

### PR DESCRIPTION
**Pull request by tomschr**
@schaefi I've looked at the main (HTML) page and think there are some information missing and we can do better. Here is my suggestion! :smiley: 

Changes proposed in this pull request:

Add QuickStart Guide and improve documentation

* Added Quick Start Guide, taken from README.md
* Improve `index.rst` file:
  * Reference quickstart in main file
  * Shortend main entry page
  * Make more headings
  * Add feature highlights to draw attention
  * Add sidebar with important KIWI links
  * Use ordered list of KIWI concept (prep and creation step)

Let me know what you think.

### CODE REVIEW

**Comment by tomschr** [comment]
Oh, and by the way: I would probably cut the README.md into a very short version or link to the guide.
Does that make sense?

File: doc/source/quickstart.rst [hunk 1]
```diff
@@ -0,0 +1,422 @@
+KIWI Quickstart
+===============
+
```
Comment (schaefi) [inline doc/source/quickstart.rst]: So we call this chapter Quickstart and within the chapter we have a topic also named Quickstart, isn't that confusing ? I think there is much more information in here than just a quick start. It feels more like an introduction. Thoughts ?

File: doc/source/quickstart.rst [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+|Build Status| |Health|
+
```
Comment (schaefi) [inline doc/source/quickstart.rst]: Does that work in the ReST source, meaning will it show the buildstatus and code health ?

File: doc/source/quickstart.rst [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+build that with the legacy KIWI, unless the appliance description
+properly encapsulates the differences into a profile.
+
+The next generation KIWI also provides the :option:`--compat` option and
+the :command:`kiwicompat` tool to be able to use the same commandline
+as provided with the legacy KIWI version.
+
+Quick Start
+-----------
+
```
Comment (schaefi) [inline doc/source/quickstart.rst]: Here is the second Quick Start I was talking about in the comment above

**Comment by schaefi** [comment]
I agree if we move everything to the sphinx documentation we should not repeat ourselves and just add a link in the README.md to the docs

File: doc/source/quickstart.rst (hunk 1 above)
Comment (tomschr) [inline doc/source/quickstart.rst]: Although I reviewed it, I missed that part obviously. I corrected the 2nd Quickstart into "Example Appliance Descriptions".

Well, basically, my idea was to convert the content inside the `README.md` into RST and include it into the other documentation.

File: doc/source/quickstart.rst (hunk 2 above)
Comment (tomschr) [inline doc/source/quickstart.rst]: I'll remove that.

File: doc/source/quickstart.rst (hunk 3 above)
Comment (tomschr) [inline doc/source/quickstart.rst]: Yes, replace with "Example Appliance Descriptions".

**Comment by tomschr** [comment]
Ok, I've committed the changes.

Apart from that, probably the quick start can be improved further. My idea was just to present the basic idea and if you like it, I can further improve. :)

**Comment by tomschr** [comment]
@schaefi After reading a bit more, maybe it's better to move the "Motivation" section of the (now) Quick Start into the `index.rst`. I think, that belongs better to an introduction/main page, than into a Quick Start.

Would that make sense?

**Comment by schaefi** [comment]
Makes all sense :smile:

**Comment by tomschr** [comment]
After reading the documentation, I thought it would be better to focus to its target group. As the text was taken from the README, the "Quick Start" guide is therefore renamed and has a more developer focused topic. I guess, it would make sense to add another Quick Start only dedicated just for *building* images (maybe also written as a tutorial).

As such, I've moved some sections from the quick start into the main `index.rst` file like the supported distributions, deprecated features etc. I think they make more sense in the main file.

I guess this can be improved, revised, and reviewed even further, but for the time being I think it's good enough. :smiley: 

Let me know if it's ok for you (should I squash all the commits later?)

**Comment by schaefi** [comment]
ok that looks nice, thanks a ton :+1:

**Pull request merged**